src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -324,7 +324,30 @@ class AppsController(
 
   private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = ???
 
-  private def killTasks(appId: PathId)(implicit identity: Identity): Route = ???
+  private def killTasks(appId: PathId)(implicit identity: Identity): Route = {
+    // the line below doesn't look nice but it doesn't compile if we use parameters directive
+    (forceParameter & parameter("host") & extractTaskKillingMode) {
+      (force, host, mode) =>
+        def findToKill(appTasks: Seq[Instance]): Seq[Instance] = {
+          appTasks.filter(_.agentInfo.host == host || host == "*")
+        }
+        mode match {
+          case TaskKillingMode.Scale =>
+            val deploymentPlanF = taskKiller.killAndScale(appId, findToKill, force)
```

Comment:
What is the F pattern? Du you mean `deploymentPlan` is a future? Why not use the type `deploymentPlan: Future[???]`?

Reply:
Yep, I do. Usually it's useful if we use maps/flatmaps so we can write like this:
```scala
val valueF = fetchValue
valueF.map { value =>
  ...
}
```